Actions/AL-Go-Helper.ps1

```diff
@@ -2026,3 +2027,35 @@ function Determine-ArtifactUrl {
     }
     return $artifactUrl
 }
+
+function Retry-Command {
+    param(
+        [Parameter(Mandatory = $true)]
+        [ScriptBlock]$Command,
+        [Parameter(Mandatory = $true)]
+        [int]$MaxRetries,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryDelaySeconds = 5
+    )
+
+    $retryCount = 0
+    while ($retryCount -lt $MaxRetries) {
+        try {
+            & $Command
+            if ($LASTEXITCODE -ne 0) {
```

Comment:
isn't LASTEXITCODE only for shell commands - is that the right error check?

Reply:
It's both for [powershell scripts and native programs](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-5.1). In this case, AzureSignTool is an executable that exits with an exitcode. That's how I capture errors in signing